docs/e2e/mobile-e2e-guidelines.md

```diff
@@ -0,0 +1,550 @@
+  // Check that Start Exploring CTA is visible & tap it
+  await TestHelpers.waitAndTap('start-exploring-button');
+  // Check that we are on the metametrics optIn screen
+  await TestHelpers.checkIfVisible('metaMetrics-OptIn');
+});
+```
+
+### 🏷️ Meaningful Naming
+
+Good:
+
+✅ Choose descriptive names for page object properties and methods that accurately convey their purpose. For example:
+
+```javascript
+class SettingsPage {}
+```
+
+❌ Bad: Unclear or ambiguous names that make it difficult to understand the purpose of the page object:
+
+```javascript
+class Screen2 {}
+```
+
+### 📦 Encapsulation of Interactions
+
+By encapsulating interaction logic within methods, test code does not need to concern itself with the specifics of how interactions are performed. This allows for easier modification of interaction behavior without impacting test scripts.
+
+✅ Good: Encapsulate interactions with page elements within methods, abstracting away implementation details. For example:
+
+```javascript
+class SettingsPage {
+  async tapNetworksButton() {
+    /*...*/
+  }
+}
+```
+
+❌ Bad: Exposing implementation details and directly interacting with elements in test code:
+
+```javascript
+// Test code
+await TestHelpers.waitAndTap('start-exploring-button');
+```
+
+### ⚙️ Utilization of Utility Functions
+
+Using utility functions is a reliable way to interact with page elements consistently across various tests. By centralizing the interaction logic and removing low-level details, utility functions improve readability and enhance the scalability of tests.
+
+✅ Good: Leverage utility functions to interact with page elements consistently across tests:
+
+```javascript
+import Matchers from '../../utils/Matchers';
+import Gestures from '../../utils/Gestures';
+
+class SettingsPage {
+  get networksButton() {
+    return Matchers.getElementByID('ELEMENT-STRING');
+  }
+
+  async tapNetworksButton() {
+    await Gestures.waitAndTap(this.networksButton);
+  }
+}
+```
+
+_NOTE:_ Matchers and Gestures are fundamental components of our test actions.
+
+Matchers Utility Class: Handles element identification and matching logic, ensuring consistency in element location across tests. This promotes consistency and reduces code duplication.
+
+Gestures Utility Class: Manages user interactions with page elements, such as tapping, swiping, or scrolling. It enhances test readability by providing descriptive methods for common user interactions.
+
+❌ Bad: Implementing interactions directly in test code without using utility functions. For example:
+
+```javascript
+await element(by.id('ELEMENT-STRING')).tap();
+```
+
+### 🗃️ Using Getters for Element Storage
+
+In our page objects, we utilize getters for element storage instead of defining selectors within the constructor. This deliberate choice stems from the behavior of getters, which are evaluated when you access the property, not when you create the object.
```

Comment:
Inconsistent spelling - behavior / behaviour, would be nice to choose one spelling for all documents.

Reply:
Interesting point Tanya. It is worth noting that there is a slight difference in spelling between Americans and Europeans, even though the words have the same meaning. Depending on who writes the documentation, their spelling might be different. However, I am hesitant to state which spelling is better for words that have the same meaning but are spelled differently.